### PR TITLE
Ignore group expansion and other minor changes when marking database as dirty

### PR DESCRIPTION
**Pull request by louib**
Allow user to ignore group expansion.

## Motivation and context
https://github.com/keepassxreboot/keepassxc/issues/446

## How has this been tested?
Locally + unit tests.

## Types of changes
- ✅ New feature

## Checklist:
- ✅ I have read the **CONTRIBUTING** document.
- ✅ My code follows the code style of this project.
- ✅ All new and existing tests passed.

### CODE REVIEW

File: src/gui/SettingsWidgetGeneral.ui [hunk 1]
```diff
@@ -106,6 +106,13 @@
         </widget>
        </item>
        <item>
+        <widget class="QCheckBox" name="ignoreGroupExpansionCheckBox">
+         <property name="text">
+          <string>Do not mark a database as modified when groups are expanded</string>
```
Comment (droidmonkey) [inline src/gui/SettingsWidgetGeneral.ui]: Do we want to expand this definition to be "Don't mark database as modified for non-data changes (e.g., expanding groups)"? This setting could apply to other things that we implement such as sort order of columns, visible columns, etc.

Comment (louib) [inline src/gui/SettingsWidgetGeneral.ui]: @droidmonkey that's a good idea!

Comment (droidmonkey) [inline src/gui/SettingsWidgetGeneral.ui]: @louib are you planning to do this for this pr?

Comment (louib) [inline src/gui/SettingsWidgetGeneral.ui]: @droidmonkey I was planning to, yes.

Comment (louib) [inline src/gui/SettingsWidgetGeneral.ui]: @droidmonkey I updated the text for the new option. I don't think there's any other change that would qualify as non-data change at the moment. Maybe reordering groups, but then again it's not something that a user does when simply looking for a password, so I think it should still mark the db as dirty.

**Pull request merged**